### PR TITLE
feat: add xapi transformer for completion events

### PR DESCRIPTION
**Pull request by Ian2012**
**Description:** Adds an xAPI transformer for completion events

**JIRA:** Link to JIRA ticket

**Dependencies:** https://github.com/openedx/completion/pull/245

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Review by ziafazal — CHANGES_REQUESTED** [review]
@Ian2012 overall looks good.  Couple of things to take care of

1. One things we need to do is change schema of xAPI statement and put progress under `result` instead of extensions of object.
2. Documentation needs to added [here](https://github.com/openedx/event-routing-backends/blob/master/docs/event-mapping/Supported_events.rst) and [here](https://github.com/openedx/event-routing-backends/blob/master/docs/event-mapping/xAPI_mapping.rst)

File: event_routing_backends/processors/xapi/tests/fixtures/expected/edx.completion.block_completion.changed.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "account": { "name": "32e08e30-f8ae-4ce2-94a8-c2bfe38a70cb", "homePage": "http://localhost:18000" }
+  },
+  "verb": { "id": "http://adlnet.gov/expapi/verbs/progressed", "display": { "en": "progressed" } },
+  "object": {
+    "id": "http://localhost:18000/xblock/block-v1:edX+DemoX+Demo_Course+type@problem+block@7c54b16c8ed34f9f8772015178c7a175",
+    "objectType": "Activity",
+    "definition": {
+      "type": "http://id.tincanapi.com/activitytype/resource",
+      "extensions": { "https://w3id.org/xapi/cmi5/result/extensions/progress": 100.0 }
+    }
```
[the remark this answers is not in the document]
Comment (Ian2012) [reply event_routing_backends/processors/xapi/tests/fixtures/expected/edx.completion.block_completion.changed.json]: thanks, I've updated the statement cc @bmtcril

**Comment by ziafazal** [comment]
@Ian2012 could you please update the documentation files to include new events?

**Review by ziafazal — APPROVED** [review]
👍

**Comment by bmtcril** [comment]
@Ian2012 should be good to go if you squash and bump the version

**Comment by Ian2012** [comment]
I will, remember that this event is not approved yet in https://github.com/openedx/completion/pull/245

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/event-routing-backends&entry.867055334=https://github.com/openedx/event-routing-backends/pull/339&entry.1484655318=https://github.com/Ian2012&entry.752974735=2023-08-22+20:34&entry.1917517419=2023-09-22+13:34&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**